### ISSUE TITLE
[$250] [Due for payment 2025-02-25] [HOLD] Web - Attachment - Console error on preview page of Large PDF uploaded

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49819]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.40-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5006507
**Email or phone of affected tester (no customers):** applausetester+uo0925@applasue.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Log into staging.new.expensify.com and open any chat
2. Tap "+" and tap Add attachment
3. Upload "Large PDF File"

## Expected Result:
No error in console appears

## Actual Result:
A console error is shown on PDF preview page.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
[Bug6615543_1727314388572!LARGE_PDF_FILE.pdf](https://github.com/user-attachments/files/17156100/Bug6615543_1727314388572.LARGE_PDF_FILE.pdf)

https://github.com/user-attachments/assets/db6337e5-48ea-4f4b-8fe3-fa8dac7ac94d

![image (241)](https://github.com/user-attachments/assets/f706d820-4971-4025-b4f2-c8c198206e1d)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021894593784371336669</li>
        <li>Upwork Job ID: 1894593784371336669</li>
        <li>Last Price Increase: 2025-02-26</li>
    </ul>
</details>

**Comment by IuliiaHerets** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by mallenexpensify** [comment]
Bumping to weekly, will review next week since it's a console error

**Comment by mallenexpensify** [comment]
Console error, not an immediate priority

**Comment by CyberAndrii** [comment]
I've opened an issue in the upstream repo: https://github.com/mozilla/pdf.js/issues/18876

**Comment by mallenexpensify** [comment]
@CyberAndrii , did you upstream PR fix this bug? I just tried to reproduce in Chrome with the file linked in the OP and I didn't see any console errors.   Added `retest-weekly` here too

**Comment by CyberAndrii** [comment]
I can still reproduce this on staging v9.0.53-1. A new version of `pdf.js` that includes the fix will be released in about a week.

**Comment by mallenexpensify** [comment]
> A new version of pdf.js that includes the fix will be released in about a week.

Reckon we're waiting on this.  We have `retest-weekly` here so we should get a retest soon.

**Comment by mallenexpensify** [comment]
Still kicking a can here, we're starting part deux for console error cleanup so this might be picked up there
- https://github.com/Expensify/App/issues/53417

**Comment by mallenexpensify** [comment]
Waiting

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/issues/53417

likely addressed in/with above issue

**Comment by mallenexpensify** [comment]
Checking here 
- https://github.com/Expensify/App/issues/53417#issuecomment-2576602822

**Comment by mallenexpensify** [comment]
From [here](https://github.com/Expensify/App/issues/53417#issuecomment-2580464029), put on hold
>  I think we have to put it on hold again, the library which we were waiting to upgrade pdfjs-dist on their side did it but it introduces another bug which is causing crash in Expensify so I will create an issue on the lib side and wait till they fix it.

**Comment by kubabutkiewicz** [comment]
Hello, Im Jakub from Callstack and would like to help with this issue

**Comment by francoisl** [comment]
@kubabutkiewicz did you end up opening an issue in pdfjs-dist?

**Comment by kubabutkiewicz** [comment]
No, sorry I missed that, I will create an issue today

**Comment by kubabutkiewicz** [comment]
@francoisl I just test it again and seems that its already fixed and its on the main branch. Can we have someone to test it and close if yes ?

**Comment by francoisl** [comment]
Nice. The issue is already labelled with `retest-weekly` so  I think it will be tested again automatically... at some point, though I'm not exactly sure when.
Otherwise @IuliiaHerets any chance we can get a retest from the Applause team here?

**Comment by IuliiaHerets** [comment]
@francoisl Tester can reproduce the issue, build - v9.0.93-3

https://github.com/user-attachments/assets/118387f3-f970-4941-9b23-a441e68a642f

**Comment by kubabutkiewicz** [comment]
I will take a look, since pdfjs-dist was already upgraded and the fix is there for that, so for me it seems to be different problem than before.

**Comment by kubabutkiewicz** [comment]
https://github.com/Expensify/react-fast-pdf/pull/46 
I found the fix, @francoisl would you be able to review this PR ?

**Comment by mallenexpensify** [comment]
@hungvu193 can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~021894593784371336669

Test case, thx @hungvu193 
https://github.com/Expensify/Expensify/issues/474385

**Comment by mallenexpensify** [comment]
Contributor+: @hungvu193 paid $250 via Upwork 

Thx!
